base/intfuncs.jl

```diff
@@ -309,14 +309,16 @@ to_power_type(x) = convert(Base._return_type(*, Tuple{typeof(x), typeof(x)}), x)
     "\nMake x a float matrix by adding a zero decimal ",
     "(e.g., [2.0 1.0;1.0 0.0]^", p, " instead of [2 1;1 0]^", p, ")",
     "or write float(x)^", p, " or Rational.(x)^", p, ".")))
-@assume_effects :terminates_locally function power_by_squaring(x_, p::Integer)
+power_by_squaring(x_, p::Integer) = power_by_squaring_with_op(*, x_, p)
```

Comment:
How about just changing this to `power_by_squaring(x_, p::Integer; *=*)` and then the rest of the function stays the same?

Reply:
Clever! Sure that sounds fine to me. Are there any perf issues with kwargs these days for a tight mathematical op like `^`?

Reply:
don't you have to do something to make it specialize on the `mul` operation?

Reply:
I think as long as you call the function, we do specialize on function args, oscar

Reply:
k thanks. it's a bit mad, but i like it. 😁 i've applied this suggestion 👍

Reply:
Oh, sadly, `*` is not an allowed variable name:
```julia
ERROR: ParseError:
# Error @ REPL[1]:1:80
@assume_effects :terminates_locally function power_by_squaring(x_, p::Integer; *=*)
#                                                                              └┘ ── invalid identifier
```
So i've pushed up one more commit here using `mul`

Reply:
With colors it becomes a bit more obvious that this is a whitespace thing:

![image](https://github.com/JuliaLang/julia/assets/154641/60ae4737-fa3b-4179-a1c8-153b0252fb65)

Kinda funny that it's trying to parse `*=` as the identifier, but spacing things out (or parentheses or both) works.

Reply:
It's totally allowed as a variable name:
```jl
julia> * = "Hello"
"Hello"
```
... and it works if you use spaces:
```jl
julia> f(x; * = *) = x*x
f (generic function with 1 method)

julia> f(3)
9
```

Reply:
The parsing issue is that `*=` is an operator, which is why you need a space.

Reply:
`(*) = (*)` is also an interesting face

Reply:
hah, i missed that -- i just assumed i was misreading the parser's highlighting. I haven't spent much time with the new parser yet and i'm not used to it - still working mostly in 1.9. Thanks all